code/iaas/agent-instance/src/main/java/io/cattle/platform/agent/instance/dao/AgentInstanceDao.java

```diff
@@ -9,7 +9,7 @@
 
     Agent getAgentByUri(String uri);
 
-    Instance getInstanceByAgent(Agent agent);
+    Instance getInstanceByAgent(long agentId);
```

Comment:
just being paranoid - is there a chance this method is called based on instance.getAgentId() // where getAgentId can potentially return null?